proposals/gc/MVP.md

```diff
@@ -817,22 +757,14 @@ The opcode for heap types is encoded as an `s33`.
 | 0xfb20 | `i31.new` |  |
 | 0xfb21 | `i31.get_s` |  |
 | 0xfb22 | `i31.get_u` |  |
-| 0xfb40 | `ref.test_canon $t` | `$t : typeidx` |
-| 0xfb41 | `ref.cast_canon $t` | `$t : typeidx` |
-| 0xfb42 | `br_on_cast_canon $l $t` | `$l : labelidx`, `$t : typeidx` |
-| 0xfb43 | `br_on_cast_canon_fail $l $t` | `$l : labelidx`, `$t : typeidx` |
-| 0xfb51 | `ref.is_data` | |
-| 0xfb52 | `ref.is_i31` | |
-| 0xfb53 | `ref.is_array` | |
-| 0xfb59 | `ref.as_data` | |
-| 0xfb5a | `ref.as_i31` | |
-| 0xfb5b | `ref.as_array` | |
-| 0xfb61 | `br_on_data` | `$l : labelidx` |
-| 0xfb62 | `br_on_i31` | `$l : labelidx` |
-| 0xfb64 | `br_on_non_data` | `$l : labelidx` |
-| 0xfb65 | `br_on_non_i31` | `$l : labelidx` |
-| 0xfb66 | `br_on_array` | `$l : labelidx` |
-| 0xfb67 | `br_on_non_array` | `$l : labelidx` |
+| 0xfb40 | `ref.test ht` | `ht : heaptype` |
+| 0xfb41 | `ref.cast ht` | `ht : heaptype` |
+| 0xfb42 | `br_on_cast $l ht` | `$l : labelidx`, `ht : heaptype` |
+| 0xfb43 | `br_on_cast_fail $l ht` | `$l : labelidx`, `ht : heaptype` |
+| 0xfb40 | `ref.test null ht` | `ht : heaptype` |
```

Comment:
Did you mean `0xfb50..0xfb53` for this and the next three? Another good option would be the 48..4b range for the `null`-enabled set.

(You may be tempted to use the range 44..47, but that is [currently](https://docs.google.com/document/d/14xWenN0lUoHa9TiDJd8yYR8ZKjgOhgO7iEmfrAuA1_M/edit#) in use in implementations, and we have a requirement not to cause breakage in the ongoing test coverage of our prototyping partners, so repurposing those would cause some logistical overhead because we'd have to do a back-and-forth migration dance, so I'd prefer to use a range that's currently not in use. Since we'll sort/compact/cleanup the encoding space before finalizing the proposal anyway, it's acceptable to have cases of minor ugliness for the time being, such as non-contiguous ranges for similar instructions.
For reference, you can always look up what prototype implementations across the stack (J2Wasm, Kotlin-Wasm, Dart2Wasm, Binaryen, V8, and I believe Spidermonkey too) are doing at https://bit.ly/3cWcm6Q, we've been keeping that link stable for the past 2.5 years and will continue to keep it. Links to drafts for updated versions, like the one linked above, are always at the top of that main document.)